tools/device_configuration_checker.py

```diff
@@ -345,4 +349,161 @@ def _run_tests_on_know_good_image(args, device):
         return (result, result_str)
 
     except Exception as error:
+        import traceback
+        traceback.print_exc()
+        logger.error(traceback.format_exc())
         return (False, "Image Test result: " + str(error))
+
+def create_work_directory(directory):
+    """
```

Comment:
Passing the files from a datafile would be smarter, I agree. Then there could be only a single function which either links or copies things as required